### PR TITLE
Enable running PEP8 on python scripts using flake8

### PR DESCRIPTION
**Pull request by skottmckay**
**Description**: 
Enable running PEP8 checks via flake8 as part of the build if flake8 is installed.

Update scripts in \tools and \onnxruntime\python to fix errors. 
  - Excluding \onnxruntime\python\tools which needs a lot more work to be PEP8 compliant. 
  - Also excluding orttraining\tools for the same reason.

The changes to the scripts fall into a few main categories
  - Whitespace related
    - primarily fixed using autopep8
  - Unused or missing imports
  - Unused variables
  - Line length

Install flake8 as part of the static_analysis build task in the Win-CPU CI so the checks are run in one CI build.
Update coding standards doc.

**Motivation and Context**
Code consistency. Topic of coding standards for python scripts has come up multiple times recently.

### CODE REVIEW

File: docs/Coding_Conventions_and_Standards.md [hunk 1]
```diff
@@ -58,3 +58,15 @@ In order to check that all the code you expect to be covered by testing is cover
 There is a configuration file in onnxruntime\VSCodeCoverage.runsettings that can be used to configure code coverage so that it reports numbers for just the onnxruntime code. Select that file in Visual Studio via the Test menu: 'Test' -> 'Test Settings' -> 'Select Test Settings File'. 
 
 Using 'Show Code Coverage Coloring' will allow you to visually inspect which lines were hit by the tests. See <https://docs.microsoft.com/en-us/visualstudio/test/using-code-coverage-to-determine-how-much-code-is-being-tested?view=vs-2017>.
+
+## Python Code Style
+
+Please adhere to the [PEP8 Style Guild](https://www.python.org/dev/peps/pep-0008/).
```
Comment (edgchen1) [inline docs/Coding_Conventions_and_Standards.md:64]: "Guild" -> "Guide"

File: cmake/flake8.cmake [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    file(GLOB_RECURSE tools_scripts CONFIGURE_DEPENDS
+        "${REPO_ROOT}/tools/*.py"
+    )
+
+    # These also need a lot of work to make PEP8 compliant
+    # file(GLOB_RECURSE training_scripts CONFIGURE_DEPENDS
+    #     "${ORTTRAINING_ROOT}/tools/*.py"
+    # )
+
+    list(APPEND python_scripts ${python_scripts_2} ${tools_scripts})
```
Comment (edgchen1) [inline cmake/flake8.cmake:37]: how about onnxruntime/test/python/*.py?
another idea is to glob_recurse all *.py and exclude the tricky ones on a blacklist (possibly composed of additional glob_recurse results of subdirectories)

**Comment by edgchen1** [comment]
Thanks for providing some guidance on Python code style.
Is yapf usage still recommended? There are some existing references to yapf.
e.g., https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/.style.yapf, https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/ReformatSourcePython.bat, https://github.com/microsoft/onnxruntime/blob/93eb9bcfdec2e3204d6a420939a112ee10e4550a/onnxruntime/python/tools/bert/BertOnnxModelKeras.py#L350

File: cmake/flake8.cmake (hunk 2 above)
Comment (thiagocrepaldi) [reply cmake/flake8.cmake:37]: I like this idea as it makes all python files being checke by default and only the explicitly blacklisted ones are skipped. Most users will create new folders and skip/forget about adding them in this whitelist

**Comment by skottmckay** [comment]
> Thanks for providing some guidance on Python code style.
> Is yapf usage still recommended? There are some existing references to yapf.
> e.g., https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/.style.yapf, https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/ReformatSourcePython.bat,

I wasn't aware of those. Seems like it's another option for an auto-formatter like autopep8. We should make things consistent though by maybe changing the default formatting style in yapf to pep8. With that possibly we could use #noqa tags instead of #yapf.

File: cmake/flake8.cmake [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#
+# Setup running flake8 on python scripts to enforce PEP8 
+# NOTE: Currently skips the check if flake8 is not installed. PRs editing python scripts are rare
+#       so don't want to add a hard dependency on flake8 to all builds.
+#
+find_program(flake8_BIN NAMES flake8)
+if(NOT flake8_BIN)
+    message(WARNING "Could not find 'flake8' to check python scripts. Please install flake8 using pip.")
```
Comment (thiagocrepaldi) [inline cmake/flake8.cmake:8]: How about changing this to FATAL_ERROR to make sure flake8 is executed locally before user submits a long CI job, wasting lots of resource on Azure

Comment (skottmckay) [reply cmake/flake8.cmake:8]: As per the reason in the comment above I don't want to add a hard dependency for all builds on flake8 being installed. Forcing someone trying to build ORT from source to install flake8 seems unnecessarily burdensome as it's unrelated to the core product. If someone is editing python files yes the first CI build will fail but presumably they will install flake8 and their local builds from then on will have the check. A one-off cost to that developer seems a better balance.

File: cmake/flake8.cmake (hunk 2 above)
Comment (skottmckay) [reply cmake/flake8.cmake:37]: Changed to glob_recurse and filter. Added some exclusions. I'm primarily trying to setup the infrastructure here so chose some key locations to update the scripts in - the onnxruntime/test scripts weren't one of those.

**Pull request merged**